docs/user/openstack/observability.md

```diff
@@ -0,0 +1,278 @@
+              cpu: 100m
+              memory: 256Mi
+          retention: 1h
+        dashboardsEnabled: false
+        dataplaneNetwork: ctlplane
+        enabled: true
+        prometheusTls: {}
+```
+
+After saving the file and letting the change propagate, verify that you receive
+metrics in the external Prometheus.
+
+#### Set up remote-write from the OCP cluster-monitoring-operator
+
+Documented [here](https://docs.openshift.com/container-platform/4.17/observability/monitoring/configuring-the-monitoring-stack.html#configuring_remote_write_storage_configuring-the-monitoring-stack).
+
+You may want to label the cluster metrics with a cluster identifier.
+
+Optionally, since metrics will be collected externally, you can set a reduced retention for local metrics.
+
+The resulting `cluster-monitoring-config` ConfigMap should then resemble this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      retention: 1h
+      remoteWrite:
+      - url: "https://external-prometheus.example/api/v1/write"
+        writeRelabelConfigs:
+        - sourceLabels:
+          - __tmp_openshift_cluster_id__
+          targetLabel: <your_cluster_id>
+          action: replace
+        tlsConfig:
+          ca:
+            secret:
+              name: mtls-bundle
+              key: ca.crt
+          cert:
+            secret:
+              name: mtls-bundle
+              key: ocp-client.crt
+          keySecret:
+            name: mtls-bundle
+            key: ocp-client.key
+```
+
+Save it to a file named `cluster-monitoring-config.yaml`. Before applying it,
+create the secret containing the HTTPS client certificates, similar to what we
+did for RHOSO.
+
+```bash
+oc --namespace openshift-monitoring \
+    create secret generic mtls-bundle \
+        --from-file=./ca.crt \
+        --from-file=ocp-client.crt \
+        --from-file=ocp-client.key
+```
+
+Once you have created the secret, it's time to apply the cluster-monitoring configuration:
+
+```bash
+oc apply -f cluster-monitoring-config.yaml
+```
+
+Let the change propagate and verify that you receive metrics in the external
+Prometheus.
+
+
+### Querying OCP metrics together with RHOSO's
+
+As an alternative to write the different metrics to a single store, it's
+possible to query remote metric stores instead.
+
+OpenShift exposes a federation endpoint to enable queries originating from
+outside the cluster to include OpenShift cluster monitoring data. You can
+follow [these instructions][federation] to set it up.
+
+[federation]: https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/monitoring/accessing-third-party-monitoring-apis#monitoring-querying-metrics-by-using-the-federation-endpoint-for-prometheus_accessing-monitoring-apis-by-using-the-cli "OpenShift documentation: Querying metrics by using the federation endpoint for Prometheus"
+
```

Comment:
~~~suggestion

#### Step 1: Gather credentials and coordinates

While connected to the OpenShift cluster through a username identified by password (as opposed to logging in using the `kubeconfig` file generated by the installer), fetch a token:

```bash
TOKEN=$(oc whoami -t)
```

Then get the Prometheus federation route URL:

```bash
HOST=$(oc -n openshift-monitoring get route prometheus-k8s-federate -ojsonpath={.status.ingress[].host})
```

#### Let RHOSO scrape OpenShift's federation endpoint

It is recommended to limit scraping to fewer than 1000 samples for each request, and with a maximum frequency of once every 30 seconds.

In this example, we will request two metrics: `kube_node_info` and `kube_persistentvolume_info`.

While connected to the RHOSO cluster, apply this manifest:

```yaml
apiVersion: monitoring.rhobs/v1alpha1
kind: ScrapeConfig
metadata:
  labels:
    service: metricStorage
  name: sos1-federated
  namespace: openstack
spec:
  params:
    'match[]':
    - '{__name__=~"kube_node_info|kube_persistentvolume_info"}'
  metricsPath: '/federate'
  authorization:
    type: Bearer
    credentials:
      name: ocp-federated
      key: token
  scheme: HTTPS # or HTTP
  scrapeInterval: 30s
  staticConfigs:
  - targets:
    - prometheus-k8s-federate-openshift-monitoring.apps.openshift.example # This is the $HOST variable fetched previously
  # add a tlsConfig stanza if the endpoint is HTTPS and if using a custom CA
```

Don't forget to make the token available as a secret (in the example above, the name is `ocp-federated`):

```bash
-n openstack ceate secret generic ocp-federated --from-literal=token=sha256~xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```
~~~